render/container.go

```diff
@@ -79,9 +79,19 @@ func (r containerWithImageNameRenderer) Render(rpt report.Report, dct Decorator)
 		if !ok {
 			continue
 		}
+		imageName, ok := image.Latest.Lookup(docker.ImageName)
+		if !ok {
+			continue
+		}
+		imageNameWithoutVersion := docker.ImageNameWithoutVersion(imageName)
+		imageNodeID := report.MakeContainerImageNodeID(imageNameWithoutVersion)
+
 		output := c.Copy()
 		output = propagateLatest(docker.ImageName, image, output)
 		output = propagateLatest(docker.ImageLabelPrefix+"works.weave.role", image, output)
+		output.Parents = output.Parents.
+			Delete(report.ContainerImage).
```

Comment:
But yes, it was for backwards compatibility - containers are reported with a parent of image id, and here we change them to be image name (the point of this PR is to key images by name in the rendering pipeline).